### PR TITLE
add filter support to phase space plugin

### PR DESCRIPTION
**Pull request by psychocoderHPC**
- remove `PhaseSpaceMulti`
- update `Cell2Particle` to support filter
  - this also requires an update in ParticleMerger
- update documentation

# Todo

- [x] tests
- [x] update documentation
- [x] merge bug fix #2428 by hand (rebase after it is merged)

### CODE REVIEW

File: docs/source/usage/plugins/phaseSpace.rst [hunk 1]
```diff
@@ -24,15 +24,17 @@ Example for *y-pz* phase space for the *electron* species (``.cfg`` file macro):
...
-====================================== ========================================= ============================
-``--e_phaseSpace.period <N>``          calculate each N steps                    *none*
-``--e_phaseSpace.space <x/y/z>``       spatial coordinate of the 2D phase space  *none*
-``--e_phaseSpace.momentum <px/py/pz>`` momentum coordinate of the 2D phase space *none*
-``--e_phaseSpace.min <ValL>``          minimum of the momentum range             :math:`m_\mathrm{species} c`
-``--e_phaseSpace.max <ValR>``          maximum of the momentum range             :math:`m_\mathrm{species} c`
-====================================== ========================================= ============================
+====================================== ======================================================== ============================
+``--e_phaseSpace.period <N>``          calculate each N steps                                   *none*
+``--e_energy.filter``                  Use filtered particles. Available filters are set up in  *none*
```
Comment (ax3l) [inline docs/source/usage/plugins/phaseSpace.rst]: typo:
- `e_energy`

File: include/picongpu/plugins/PhaseSpace/PhaseSpace.hpp [hunk 2]
```diff
@@ -41,6 +43,7 @@
 
 #include <string>
 #include <utility>
+#include <mpi.h>
```
Comment (ax3l) [inline include/picongpu/plugins/PhaseSpace/PhaseSpace.hpp]: If I remember correctly, MPI is quite sensitive to be included *before* standard libs...?
(I might also be too old already and this is a flash back memory from the time I was still young and naïve)

File: docs/source/usage/plugins/phaseSpace.rst (hunk 1 above)
Comment (psychocoderHPC) [reply docs/source/usage/plugins/phaseSpace.rst]: Note to my self: do not copy past code snippets

File: include/picongpu/plugins/PhaseSpace/PhaseSpace.hpp (hunk 2 above)
Comment (psychocoderHPC) [reply include/picongpu/plugins/PhaseSpace/PhaseSpace.hpp]: No there is no need to place the MPI include to the top. Even if the include is the first, includes from other files will be on top of the MPI include.

Comment (ax3l) [reply include/picongpu/plugins/PhaseSpace/PhaseSpace.hpp]: true.

**Comment by psychocoderHPC** [comment]
mErGe mE If YoU cAn

**Comment by theZiz** [comment]
mErGe mE If YoU cAn

**Pull request merged**